nav2_collision_monitor/include/nav2_collision_monitor/types.hpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2022 Samsung Research Russia
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_COLLISION_MONITOR__TYPES_HPP_
+#define NAV2_COLLISION_MONITOR__TYPES_HPP_
+
+namespace nav2_collision_monitor
+{
+
+/// @brief Velocity for 2D model of motion
+struct Velocity
```

Comment:
If you add a member function `isZero()` then that cleans up the logic in the node's check if the current requested speed is zero (e.g. `if (velocity.isZero()) {...}`)

Reply:
Actually in the second nest too if the previous is non-zero! That would make it look much nicer